### PR TITLE
Fix: Removed ShardedAccount

### PR DESCRIPTION
**Pull request by egor0798**
Remove `ShardedAccount` and generate account structure

### CODE REVIEW

File: src/renderer/shared/core/types/account.ts [hunk 1]
```diff
@@ -29,7 +24,7 @@ export type ChainAccount = BaseAccount & {
 };
 
 export type ShardAccount = BaseAccount & {
-  shardedId: ID;
+  group: ID;
```
Comment (sokolova-an) [inline src/renderer/shared/core/types/account.ts:27]: we also decided to add keyType to ShardAcсount, right?

agree with your message in tg - `groupId` sounds better here

File: src/renderer/entities/wallet/lib/account-utils.ts [hunk 2]
```diff
@@ -83,3 +78,25 @@ function getAllAccountIds(accounts: Account[], chainId: ChainId): AccountId[] {
 function getWalletAccounts<T extends Account>(walletId: Wallet['id'], accounts: T[]): T[] {
   return accounts.filter((account) => account.walletId === walletId);
 }
+
+function getAccountsWithGroupedShards<T extends Account>(accounts: T[]): Array<T | Array<T>> {
+  const shardsIndexes: Record<ShardAccount['group'], number> = {};
```
Comment (sokolova-an) [inline src/renderer/entities/wallet/lib/account-utils.ts:83]: maybe use `Set` here?

in this case, we can check the following conditions with: `shardsSet.has(account.group)`

**Review by sokolova-an — COMMENTED** [review]
good job!

File: src/renderer/shared/core/types/account.ts (hunk 1 above)
Comment (tuul-wq) [reply src/renderer/shared/core/types/account.ts:27]: yeah, `groupId` is a nice one

File: src/renderer/entities/wallet/lib/account-utils.ts [hunk 3]
```diff
@@ -83,3 +78,25 @@ function getAllAccountIds(accounts: Account[], chainId: ChainId): AccountId[] {
 function getWalletAccounts<T extends Account>(walletId: Wallet['id'], accounts: T[]): T[] {
   return accounts.filter((account) => account.walletId === walletId);
 }
+
+function getAccountsWithGroupedShards<T extends Account>(accounts: T[]): Array<T | Array<T>> {
```
Comment (tuul-wq) [inline src/renderer/entities/wallet/lib/account-utils.ts:82]: Please add test

Comment (tuul-wq) [inline src/renderer/entities/wallet/lib/account-utils.ts:82]:
```suggestion
function getAccountsAndShardGroups<T extends Account>(accounts: T[]): Array<T | Array<T>> {
```

**Review by tuul-wq — APPROVED** [review]
Approved with a couple of comments 👍

File: src/renderer/shared/core/types/account.ts (hunk 1 above)
Comment (tuul-wq) [reply src/renderer/shared/core/types/account.ts:27]: Also not sure about simple `ID`, we discussed `uuid` so it won't intersect with other `groupId's`

File: src/renderer/entities/wallet/lib/account-utils.ts (hunk 2 above)
Comment (Asmadek) [reply src/renderer/entities/wallet/lib/account-utils.ts:83]: Good solution `set.has` has `O(1)` time complexity, but I think here we store indexes for shards array, and can't use such solution

**Review by Asmadek — APPROVED** [review]

**Review by sokolova-an — APPROVED** [review]

File: src/renderer/shared/core/types/account.ts (hunk 1 above)
Comment (egor0798) [reply src/renderer/shared/core/types/account.ts:27]: changed groupId to string and added keyType

File: src/renderer/entities/wallet/lib/account-utils.ts (hunk 2 above)
Comment (egor0798) [reply src/renderer/entities/wallet/lib/account-utils.ts:83]: yeah, we need pairs of groupId - index so we cant use set here

File: src/renderer/entities/wallet/lib/account-utils.ts (hunk 3 above)
Comment (egor0798) [reply src/renderer/entities/wallet/lib/account-utils.ts]: done

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**